src/jit/lowerxarch.cpp

```diff
@@ -842,77 +842,91 @@ void Lowering::LowerSIMD(GenTreeSIMD* simdNode)
-            // is expected to set the Zero flag appropriately.
-            // All the involved nodes must form a continuous range, there's no other way to
-            // guarantee that condition flags aren't changed between the SIMD node and the JCC
-            // node.
-            //
-
-            bool     transformed = false;
-            GenTree* simdUser    = simdUse.User();
+        simdNode->gtType = TYP_VOID;
+        simdNode->ClearUnusedValue();
+    }
+#endif
+    ContainCheckSIMD(simdNode);
+}
+#endif // FEATURE_SIMD
 
-            if (simdUser->OperIs(GT_EQ, GT_NE) && simdUser->gtGetOp2()->IsCnsIntOrI() &&
-                (simdNode->gtNext == simdUser->gtGetOp2()) && (simdUser->gtGetOp2()->gtNext == simdUser))
-            {
-                ssize_t relopOp2Value = simdUser->gtGetOp2()->AsIntCon()->IconValue();
+#ifdef FEATURE_HW_INTRINSICS
 
-                if ((relopOp2Value == 0) || (relopOp2Value == 1))
-                {
-                    GenTree* jtrue = simdUser->gtNext;
+//----------------------------------------------------------------------------------------------
+// LowerHWIntrinsicCC: Lowers a hardware intrinsic node that produces a boolean value by
+//     setting the condition flags.
+//
+//  Arguments:
+//     node - The hardware intrinsic node
+//     newIntrinsicId - The intrinsic id of the lowered intrinsic node
+//     condition - The condition code of the generated SETCC/JCC node
+//
+void Lowering::LowerHWIntrinsicCC(GenTreeHWIntrinsic* node, NamedIntrinsic newIntrinsicId, GenCondition condition)
+{
+    GenTreeCC* cc = LowerNodeCC(node, condition);
 
-                    if ((jtrue != nullptr) && jtrue->OperIs(GT_JTRUE) && (jtrue->gtGetOp1() == simdUser))
-                    {
-                        if ((simdNode->gtSIMDIntrinsicID == SIMDIntrinsicOpEquality) != simdUser->OperIs(GT_EQ))
-                        {
-                            relopOp2Value ^= 1;
-                        }
+    node->gtHWIntrinsicId = newIntrinsicId;
+    node->gtType          = TYP_VOID;
+    node->ClearUnusedValue();
 
-                        jtrue->ChangeOper(GT_JCC);
-                        GenTreeCC* jcc = jtrue->AsCC();
-                        jcc->gtFlags |= GTF_USE_FLAGS;
-                        jcc->gtCondition = (relopOp2Value == 0) ? GenCondition::NE : GenCondition::EQ;
+    bool swapOperands    = false;
+    bool canSwapOperands = false;
 
-                        BlockRange().Remove(simdUser->gtGetOp2());
-                        BlockRange().Remove(simdUser);
-                        transformed = true;
-                    }
-                }
+    switch (newIntrinsicId)
+    {
+        case NI_SSE_COMISS:
+        case NI_SSE_UCOMISS:
+        case NI_SSE2_COMISD:
+        case NI_SSE2_UCOMISD:
+            // Using the preferred condition saves a branch so it's probably better
+            // to always use it, even if that means loosing containment.
```

Comment:
I am missing where this handles the case where it needs to lose containment

Reply:
If we have `[mem], reg` we may be able to contain the mem operand by swapping the operands to have `reg, [mem]`. But that also requires changing the comparison condition and unfortunately some FP conditions have complicated codegen on xarch. So the code below gives priority to "preferrable" FP conditions by blocking operands swapping and thus indirectly blocking containment in some cases.

I'll try to expand this comment a bit tomorrow, together with the other comment improvement suggestion you've made.

Reply:
I think I now see my point of confusion. When we get here, we haven't yet done containment analysis, correct? That's why we don't need to "undo" anything.

Reply:
Yes. Hopefully the new comments is more clear.